engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -44,6 +44,9 @@ final case class ExecutionStore(private val statusStore: Map[JobKey, ExecutionSt
     (key.scope.fullyQualifiedName, key.index) -> key
   } toMap
 
+  lazy val terminalKeys: Set[FqnIndex] = store.filterKeys(_.isTerminal).values.flatten.map { key =>
```

Comment:
ToL: To avoid going through the store again (since we already do it just above for `doneKeys`), we could do it all at once with something like

```
  val (doneKeys, terminalKeys) = {
    def toMapEntry(key: JobKey) = (key.scope.fullyQualifiedName, key.index) -> key
    
    store.foldLeft((Map.empty[FqnIndex, JobKey], Map.empty[FqnIndex, JobKey]))({
      case ((done, terminal), (status, keys))  =>
        val newMapEntries = keys map toMapEntry
        val newDone = if (status.isDoneOrBypassed) done ++ newMapEntries else done
        val newTerminal = if (status.isTerminal) terminal ++ newMapEntries else terminal
        
        newDone -> newTerminal
    })
  }
```

Although this is not really critical because this store only has as many entries as we have different ExecutionStatus values which is not a lot.

I would still run `ExecutionStoreBenchmark` with and without your changes to see if it has an impact :)